core/registry/servicecenter/cache.go

```diff
@@ -225,10 +225,6 @@ func (c *CacheManager) getServiceStore(exist []*model.MicroService) sets.String
 func filterReIndex(providerInstances []*model.MicroServiceInstance, serviceName string, appID string) {
 	var store = make([]*registry.MicroServiceInstance, 0, len(providerInstances))
 	for _, ins := range providerInstances {
-		if ins.Status != model.MSInstanceUP {
```

Comment:
you commend here and algorithm complexity is O(n) to just get unified slice, but you can actually separate into two slice, then you don't need another  O(n) to just figure out what is not UP status again